### ISSUE TITLE
How to rollback optimizer step (i.e. reload best checkpoint) during training?

### ISSUE THREAD

**Issue by xsys-technology** [PyTorchLightning/pytorch-lightning#13099]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/13093

<div type='discussions-op-text'>

<sup>Originally posted by **xsys-technology** May 17, 2022</sup>
During training, if the optimizer takes a really bad step that sends validation loss through the roof, how can one rollback the step and reload best model checkpoint (and reset logging results/metrics so that _val_loss_ remains accurate)? 

I imagine the preferred way to do this is via a callback function (e.g. '_on_train_epoch_start_') but I'm not sure how to do this properly/safely when the model is dispatched and sharded across gpu's.</div>
